core/src/main/java/com/exadel/aem/backpack/core/services/impl/PackageServiceImpl.java

```diff
@@ -304,29 +318,150 @@ public PackageInfo createPackage(final ResourceResolver resourceResolver, final
+            packageInfos.asMap().remove(modificationPackageModel.getPackagePath());
+            packageInfos.asMap().put(packageInfo.getPackagePath(), packageInfo);
+        }
+
+        return packageInfo;
+    }
+
+    /**
+     * Called from {@link PackageService#editPackage(ResourceResolver, PackageModel)} to check whether
+     * properties that affect package location were updated during modification
+     *
+     * @param newPkg {@link PackageModel} model with package modification info
+     * @param oldPkg {@link PackageModel} model with existing package info
+     * @return {@code boolean}
+     */
+    private boolean isPackageLocationUpdated(final PackageModel newPkg, final PackageModel oldPkg) {
+        return !oldPkg.getPackageName().equals(newPkg.getPackageName()) ||
+                !oldPkg.getGroup().equals(newPkg.getGroup()) ||
+                (StringUtils.isNotBlank(oldPkg.getVersion()) && StringUtils.isNotBlank(newPkg.getVersion()) && !oldPkg.getVersion().equals(newPkg.getVersion()));
+    }
+
+    /**
+     * Called from {@link PackageServiceImpl#editPackage(ResourceResolver, PackageModel)} in order to update package location and information
+     *
+     * @param userSession Current user {@code Session} as adapted from the acting {@code ResourceResolver}
+     * @param packagePath Modified package path
+     * @param packageInfo {@code PackageInfo} object to store status information in
+     * @param paths       {@code List} of {@code PathModel} will be stored in package metadata information and used in future package modifications
+     * @param filter      {@code DefaultWorkspaceFilter} instance representing resource selection mechanism for the package
+     */
+    private void modifyPackage(final Session userSession,
+                               final String packagePath,
+                               final PackageInfo packageInfo,
+                               final List<PathModel> paths, final DefaultWorkspaceFilter filter) {
+        JcrPackage jcrPackage = null;
+        try {
+            JcrPackageManager packMgr = PackagingService.getPackageManager(userSession);
+            if (!filter.getFilterSets().isEmpty()) {
+                Node packageNode = userSession.getNode(packagePath);
+                if (packageNode != null) {
+                    jcrPackage = packMgr.open(packageNode);
+                    jcrPackage = packMgr.rename(jcrPackage, packageInfo.getGroupName(), packageInfo.getPackageName(), packageInfo.getVersion());
+                    JcrPackageDefinition jcrPackageDefinition = jcrPackage.getDefinition();
+                    if (jcrPackageDefinition != null) {
+                        setPackageInfo(userSession, packageInfo, paths, filter, jcrPackageDefinition);
+                        packageInfo.setPackageStatus(PackageStatus.MODIFIED);
+                        packageNode = jcrPackage.getNode();
+                        if (packageNode != null) {
```

Comment:
Thanks! Method simplified. To be honest I prefer nested conditions(to some point) over the multiple returns because it's possible to extract sub-methods while with multiple returns, it's not.